### ISSUE TITLE
Linter always invoked on a temporary file?

### ISSUE THREAD

**Issue by Hibou57** [SublimeLinter/SublimeLinter-phplint#18]
Unless I overlooked something (I really looked at the doc), it seems `SublimeLinter-phplint` is always invoked on a temporary file, which unfortunately makes it unusable when PHP files `require_once` some files whose location are relative to depending files' own location (using `__DIR__`).

Searching the doc, I could find about a [tempfile-suffix](http://www.sublimelinter.com/en/latest/linter_attributes.html#tempfile-suffix), but that's a plugin attribute, which seemingly cannot be set in preferences.

I don't know what this value is for `SublimeLinter-phplint`, but I guess it should be “-” and it is not.

I initially though this was due to the SublimeLinter running in background mode, but setting it so that it runs in load/save mode, did not solve anything.

I don't know if phplint can work on standard input, which would be the best (would be OK with background mode).

**Comment by Hibou57** [comment]
At least, here is a workaround, involving a patched `phppl`.

Prior to anything, for it to work, ` "chdir": "${directory}"` is assumed. Ex:

    "linters": {
        "phplint": {
            "@disable": false,
            "args": [],
            "chdir": "${directory}",
            "excludes": []
        },
        "pylint": {
            ....
        },
    },

In `phppl`, I changed this:

	"$__DIR__/php" \
		"$__DIR__/stdlib/it/icosaedro/lint/PHPLint.php" \
		--modules-path "$__DIR__/modules" \
		--php-version 5 \
		--print-path relative \
		--print-errors \
		--print-warnings \
		--print-notices \
		--ascii-ext-check \
		--ctrl-check \
		--recursive \
		--no-print-file-name \
		--parse-phpdoc \
		--print-context \
		--print-source \
		--print-line-numbers \
		--report-unused \
		"$@"

Into this (use a bit of Bash hacking):

	# "${@: -1}" is last command line argument
	# "${@:1:$#-1}" is all command line arguments but #0 and the last

	ORG_TMP_FILE="${@: -1}";
	NEW_TMP_FILE_PATTERN="XXXXX-$(basename $ORG_TMP_FILE)";
	NEW_TMP_FILE=$(mktemp "$NEW_TMP_FILE_PATTERN")
	cp "$ORG_TMP_FILE" "$NEW_TMP_FILE";

		"$__DIR__/php" \
			"$__DIR__/stdlib/it/icosaedro/lint/PHPLint.php" \
			--modules-path "$__DIR__/modules" \
			--php-version 5 \
			--print-path relative \
			--print-errors \
			--print-warnings \
			--print-notices \
			--ascii-ext-check \
			--ctrl-check \
			--recursive \
			--no-print-file-name \
			--parse-phpdoc \
			--print-context \
			--print-source \
			--print-line-numbers \
			--report-unused \
			"${@:1:$#-1}" \
			"$NEW_TMP_FILE";

	rm "$NEW_TMP_FILE";

Basically, it copies the file in `/tmp` to a new temporary file in the same directory as that of the original source file (not the one in `/tmp`), passed all arguments but the last, then for the last, pass the newly crafted one. Finally, the additional temporary file is removed.

**Comment by Hibou57** [comment]
Oops, the above was too naive. At startup, `phpl` is invoked to request its version. So the above is correct only if the last argument is indeed a file, otherwise, it needs to be invoked as before (I know it's ugly, I'm not a Bash hacker):

	# "${@: -1}" is last command line argument
	# "${@:1:$#-1}" is all command line arguments but #0 and the last

	ORG_TMP_FILE="${@: -1}";

	if [ -f "$ORG_TMP_FILE" ]; then

		NEW_TMP_FILE_PATTERN="tmp-XXXXX-$(basename $ORG_TMP_FILE)";
		NEW_TMP_FILE=$(mktemp "$NEW_TMP_FILE_PATTERN");
		cp "$ORG_TMP_FILE" "$NEW_TMP_FILE";

		"$__DIR__/php" \
			"$__DIR__/stdlib/it/icosaedro/lint/PHPLint.php" \
			--modules-path "$__DIR__/modules" \
			--php-version 5 \
			--print-path relative \
			--print-errors \
			--print-warnings \
			--print-notices \
			--ascii-ext-check \
			--ctrl-check \
			--recursive \
			--no-print-file-name \
			--parse-phpdoc \
			--print-context \
			--print-source \
			--print-line-numbers \
			--report-unused \
			"${@:1:$#-1}" \
			"$NEW_TMP_FILE";

		rm "$NEW_TMP_FILE";
	else
		"$__DIR__/php" \
			"$__DIR__/stdlib/it/icosaedro/lint/PHPLint.php" \
			--modules-path "$__DIR__/modules" \
			--php-version 5 \
			--print-path relative \
			--print-errors \
			--print-warnings \
			--print-notices \
			--ascii-ext-check \
			--ctrl-check \
			--recursive \
			--no-print-file-name \
			--parse-phpdoc \
			--print-context \
			--print-source \
			--print-line-numbers \
			--report-unused \
			"$@";
	fi;
